05/app.js

```diff
@@ -12,76 +11,75 @@ function handleSubmit(e) {
+        { name: 'flatNumber', label: 'Numer mieszkania', type: 'number', required: false },
+        { name: 'zip', label: 'Kod pocztowy', pattern: '[0-9]{2}-[0-9]{3}', required: true },
+        { name: 'city', label: 'Miejscowość', required: true },
+        { name: 'voivodeship', label: 'Województwo', required: true }]
 
-    for (let i = 0; i < formElements.length; i++) {
-        const formElement = formElements[i]
-        resetInputAndLabel(formElement)
-        checkCondition(formElement)
-    }
-    checkFormValidation()
+    fields.forEach(function (field) {
+        const formElement = form.elements[field.name]
+        resetMarkedElements(formElement)
+        checkCondition(formElement, field)
+    })
+
+    checkFormValidation(fields)
 }
 
 function resetMessagesList() {
     const messagesList = document.querySelector('.messages')
     messagesList.innerHTML = ' '
 }
 
-function resetInputAndLabel(element) {
+function resetMarkedElements(element) {
     element.parentElement.style.color = 'black'
     element.style.borderColor = "black"
 }
 
-function checkCondition(element) {
-    if (element.name == "zip") {
-        const pattern = /[0-9]{2}-[0-9]{3}/
-        if (!pattern.test(element.value)) {
-            const elementName = element.parentElement.textContent
-            const error = `Pole "${elementName}" nie może być puste. Wymagany format: XX-XXX (X-cyfra)`
-            createErrorItem(error, element)
-        }
-    } else if (element.type === 'text') {
-        if (!(element.value.trim().length > 0)) {
-            const elementName = element.parentElement.textContent
-            const error = `Pole "${elementName}" nie może być puste. `
+function checkCondition(element, field) {
+    if (field.required) {
+        if (element.value.length === 0) {
+            const error = `Pole "${field.label}" nie może być puste. `
             createErrorItem(error, element)
         }
-    } else if (element.name === 'houseNumber') {
-        if (element.value <= 0 || element.value.trim().length === 0) {
-            const elementName = element.parentElement.textContent
-            const error = `Pole "${elementName}" nie może być puste i musi być większe od 0.`
-            createErrorItem(error, element)
-        }
-    } else if (element.name === 'flatNumber') {
+    }
+    if (field.type === 'number') {
         if (element.value < 0 || element.value === '0') {
-            const elementName = element.parentElement.textContent
-            const error = `Pole "${elementName}" musi być większe od 0.`
+            const error = `Wartość w polu "${field.label}" musi być większa od 0. `
             createErrorItem(error, element)
         }
-    } else if (element.tagName === "SELECT") {
-        if (element.selectedIndex === 0) {
-            const elementName = element.parentElement.innerText.split(' ')[0]
-            const error = `Pole "${elementName}" musi zostać wybrane.`
+    }
+    if (field.name === "zip") {
```

Comment:
Tutaj jest na sztywno. Raczej bym sprawdzał czy jest zdefiniowany `pattern` - jeśli tak to go zastosuj czyli
```
 if (field.pattern) {
     // sprawdź czy się zgadza i ewentualnie dodaj komunikat
 }
```